ddtrace/opentelemetry/span_test.go

```diff
@@ -22,63 +24,248 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/httpmem"
 )
 
-func TestSpanMethods(t *testing.T) {
-	testData := struct {
-		env, srv, oldOp, newOp string
-		tags                   [][]string // key - tags[0][0], value - tags[0][1]
-	}{
-		env:   "test_env",
-		srv:   "test_srv",
-		oldOp: "old_op",
-		newOp: "new_op",
-		tags:  [][]string{{"tag_1", "tag_1_val"}, {"opt_tag_1", "opt_tag_1_val"}}}
-	done := make(chan struct{})
-	var payload string
+func mockTracerProvider(t *testing.T, opts ...tracer.StartOption) (tp *TracerProvider, payloads chan string, cleanup func()) {
+	payloads = make(chan string)
 	s, c := httpmem.ServerAndClient(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Path {
 		case "/v0.4/traces":
+			if h := r.Header.Get("X-Datadog-Trace-Count"); h == "0" {
+				return
+			}
 			buf, err := io.ReadAll(r.Body)
-			if err != nil {
+			if err != nil || len(buf) == 0 {
+				t.Log("Test agent: Error receiving traces")
 				t.Fail()
```

Comment:
```suggestion
				t.Fatalf("mock trace agent error receiving traces: %v", err)
```
`t.Fatal` just logs and calls fail, so let's use that: https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/testing/testing.go;l=1054

Plus, if you have an error from a test, you basically always want to log it.

Reply:
(Here and throughout)